packages/perseus/src/widgets/categorizer/categorizer.tsx

```diff
@@ -40,7 +45,10 @@ type State = {
     uniqueId: string;
 };
 
-export class Categorizer extends React.Component<Props, State> {
+export class Categorizer
+    extends React.Component<Props, State>
+    implements Widget
+{
```

Comment:
It would have to be a generic type `WidgetComponent<Props, State> = React.Component<Props, State> implements Widget`. 🤔  I'm a bit concerned it'd be too restrictive.

Reply:
Ok, I tried but it feels like I'm going down a rabbit hole. I can make an abstract class:

```ts
abstract class WidgetComponent<Props, State>
    extends React.Component<Props, State>
    implements Widget
{}
```

But then it complains that no methods from Widget are implemented in this abstract class.  If I add one of the functions and mark _it_ as abstract, then any widget that hasn't implemented that function now need to implement it, because it's abstract. 

I'm going to leave it as is, but at least we explored the idea.